pkg/minikube/config/profile_test.go

```diff
@@ -277,7 +279,11 @@ func TestGetPrimaryControlPlane(t *testing.T) {
 			t.Fatalf("Failed to load config for %s", tc.description)
 		}
 
-		n, err := PrimaryControlPlane(cc)
+		if tc.description == "old style" {
```

Comment:
Is it ok to make this line rather than separating the test with each testcase itself?

Reply:
can you explain/comment what does old style mean in this context?

Reply:
`TestGetPrimaryControlPlane` have two tests:
`{"old style", "p1", "192.168.64.75", 8443, "minikube"}, {"new style", "p2_newformat", "192.168.99.136", 8443, "m01"}`

such case load config from `minikube/pkg/minikube/config/testdata/.minikube2/profiles/p1/config.json` and `minikube/pkg/minikube/config/testdata/.minikube2/profiles/p2_newformat/config.json`

But for `p1`, `PrimaryControlPlane` function create "the new version config of p1" save it with same `ProfileName`.
Therefore, the testdata `minikube/pkg/minikube/config/testdata/.minikube2/profiles/p1/config.json` updated with "the new version config of p1".
(I guess this happens since the `PrimaryControlPlane` wants to be compatible with old version(1.6 or earlier) config file where `node` was not implemented.)

So for the "old version" test case, I think we have to create "the new version config of p1" save it with different `ProfileName`.
So I changed the `ProfileName` for "old version". So the test make "the new version config of p1" at `minikube/pkg/minikube/config/testdata/.minikube2/profiles/p1_converted/config.json` and delete the file when test finished.

Reply:
I don't think the old style should require any special handling: I think the intent of this test is to assert that.